examples/js/loaders/GLTFLoader.js

```diff
@@ -1547,59 +1523,57 @@ THREE.GLTFLoader = ( function () {
 
 	}
 
-	function isArrayEqual( a, b ) {
+	function createGeometryKey( geometry ) {
 
-		if ( a.length !== b.length ) return false;
+		var dracoExtension = geometry.extensions && geometry.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ];
+		var geometryKey;
 
-		for ( var i = 0, il = a.length; i < il; i ++ ) {
+		if ( dracoExtension ) {
 
-			if ( a[ i ] !== b[ i ] ) return false;
+			geometryKey = 'draco:' + dracoExtension.bufferView
+				+ ':' + dracoExtension.indices
+				+ ':' + createAttributesKey( dracoExtension.attributes );
+
+		} else {
+
+			geometryKey = geometry.indices + ':' + createAttributesKey( geometry.attributes );
 
 		}
 
-		return true;
+		return geometryKey;
 
 	}
 
-	function getCachedGeometry( cache, newPrimitive ) {
+	function createAttributesKey( attributes ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var attributesKey = '';
 
-			var cached = cache[ i ];
+		for ( var key in attributes ) {
 
-			if ( isPrimitiveEqual( cached.primitive, newPrimitive ) ) return cached.promise;
+			attributesKey += key + attributes[ key ];
 
 		}
 
-		return null;
+		return attributesKey;
 
 	}
+	function createArrayKey( a ) {
 
-	function getCachedCombinedGeometry( cache, geometries ) {
-
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var arrayKey = '';
 
-			var cached = cache[ i ];
+		for ( var i = 0, il = a.length; i < il; i ++ ) {
 
-			if ( isArrayEqual( geometries, cached.baseGeometries ) ) return cached.geometry;
+			arrayKey += i + a[ i ];
```

Comment:
I don't think this `createArrayKey` function will work as intended — the way this method is used now, the array will contain either `THREE.Geometry` or glTF primitive definitions. In either case, the string for an array element will be `[object Object]`.

Reply:
I will save the values of the keys, but at the momentthe `isArrayEqual` just does a direct comparison between elements of the arrays, so I thought there were inside strings or something like that

Reply:
Currently `a` passed into this method could be an `Array<THREE.BufferGeometry>` or `Array<GLTF.Primitive>` here in which case we're doing something equivalent to:

```js
arrayKey += 25 + new THREE.BufferGeometry();
```

The direct comparison was possible because if the mesh had identical primitives in the same order, then earlier caching would ensure it has the same exact geometry objects.